plugins/azure/fix_plugin_azure/resource/mysql.py

```diff
@@ -318,39 +319,37 @@ def collect(
-        "type": S("type"),
-        "ctime": S("systemData", "createdAt"),
-        "mtime": S("systemData", "lastModifiedAt"),
-        "allowed_values": S("properties", "allowedValues"),
-        "current_value": S("properties", "currentValue"),
-        "data_type": S("properties", "dataType"),
-        "default_value": S("properties", "defaultValue"),
-        "description": S("properties", "description"),
-        "documentation_link": S("properties", "documentationLink"),
-        "is_config_pending_restart": S("properties", "isConfigPendingRestart") >> AsBool(),
-        "is_dynamic_config": S("properties", "isDynamicConfig") >> AsBool(),
-        "is_read_only": S("properties", "isReadOnly") >> AsBool(),
-        "source": S("properties", "source"),
-        "value": S("properties", "value"),
-    }
-    allowed_values: Optional[str] = field(default=None, metadata={'description': 'Allowed values of the configuration.'})  # fmt: skip
-    current_value: Optional[str] = field(default=None, metadata={"description": "Current value of the configuration."})
-    data_type: Optional[str] = field(default=None, metadata={"description": "Data type of the configuration."})
-    default_value: Optional[str] = field(default=None, metadata={"description": "Default value of the configuration."})
-    description: Optional[str] = field(default=None, metadata={"description": "Description of the configuration."})
-    documentation_link: Optional[str] = field(default=None, metadata={'description': 'The link used to get the document from community or Azure site.'})  # fmt: skip
-    is_config_pending_restart: Optional[bool] = field(default=None, metadata={'description': 'If is the configuration pending restart or not.'})  # fmt: skip
-    is_dynamic_config: Optional[bool] = field(default=None, metadata={'description': 'If is the configuration dynamic.'})  # fmt: skip
-    is_read_only: Optional[bool] = field(default=None, metadata={"description": "If is the configuration read only."})
-    source: Optional[str] = field(default=None, metadata={"description": "Source of the configuration."})
-    value: Optional[str] = field(default=None, metadata={"description": "Value of the configuration."})
-    system_data: Optional[AzureSystemData] = field(default=None, metadata={'description': 'Metadata pertaining to creation and last modification of the resource.'})  # fmt: skip
-    type: Optional[str] = field(default=None, metadata={'description': 'The type of the resource. E.g. Microsoft.Compute/virtualMachines or Microsoft.Storage/storageAccounts '})  # fmt: skip
+    config: Json = field(factory=dict)
+
+    @classmethod
+    def collect(
+        cls,
+        raw: List[Json],
+        builder: GraphBuilder,
+    ) -> List["AzureMysqlServerConfiguration"]:
+        if not raw:
+            return []
+        server_id = raw[0].get("serverID")
+        if not server_id:
+            return []
```

Comment:
Consider defining a separate class method - not collect.
It would allow you to easily pass all the relevant information from the server.